packages/aragon-wrapper/src/index.js

```diff
@@ -450,6 +469,276 @@ export default class Aragon {
       publishReplay(1)
     )
     this.apps.connect()
+
+    /*******************************
+     *                             *
+     *            REPOS            *
+     *                             *
+     ******************************/
+
+    // Initialize installed repos from the list of apps
+    const installedRepoCache = new Map()
+    const repo$ = apps$.pipe(
+      // Map installed apps into a deduped list of their aragonPM repos, with these assumptions:
+      //   - No apps are lying about their appId (malicious apps _could_ masquerade as other
+      //     apps by setting this value themselves)
+      //   - `contractAddress`s will stay the same across all installed apps.
+      //      This is technically not true as apps could set this value themselves
+      //      (e.g. as pinned apps do), but these apps wouldn't be able to upgrade anyway
+      //
+      //  Ultimately returns an array of objects, holding the repo's:
+      //    - appId
+      //    - base contractAddress
+      map((apps) => Object.values(
+        apps
+          .filter(({ appId }) => !isAragonOsInternalApp(appId))
+          .reduce((installedRepos, { appId, codeAddress, updated }) => {
```

Comment:
Here we reduce the apps array of objects into an object which is then transformed into an array again by the wrapping `Object.values` call. Can't we just map the array after filter directly into the format without the transformation into an object?

Reply:
The reason we map back and forth between the array and the object is to dedupe the array based on the `appId`, as it may include multiple instances of an app (e.g. the AGP org has two voting app instances).

This could be combined with the following `map()` operator, which may be more clear 🤷‍♂️.